ts/annotations/navigationBindings.ts

```diff
@@ -0,0 +1,1626 @@
+            verticalLine: Array<string>;
+        }
+        interface AnnotationNonEditableObject {
+            rectangle: Array<string>;
+        }
+        interface AnnotationsOptions {
+            langKey?: string;
+        }
+        interface Chart {
+            navigationBindings?: NavigationBindings;
+            /** @requires modules/annotations */
+            initNavigationBindings(): void;
+        }
+        interface LangNavigationOptions {
+            popup?: Dictionary<string>;
+        }
+        interface LangOptions {
+            navigation?: LangNavigationOptions;
+        }
+        class NavigationBindings {
+            public static annotationsEditable: AnnotationEditableObject;
+            public static annotationsNonEditable: AnnotationNonEditableObject
+            public constructor(chart: AnnotationChart, options: NavigationOptions);
+            public activeAnnotation?: (false|Annotation);
+            public boundClassNames: Dictionary<NavigationBindingsOptionsObject>;
+            public chart: AnnotationChart;
+            public container: HTMLDOMElement;
+            public currentUserDetails?: Annotation;
+            public eventsToUnbind: Array<Function>;
+            public mouseMoveEvent?: (false|Function);
+            public nextEvent?: (false|Function);
+            public options: NavigationOptions;
+            public selectedButtonElement?: (HTMLDOMElement|null);
+            public selectedButton: (NavigationBindingsOptionsObject|null);
+            public stepIndex?: number;
+            public steps?: boolean;
+            public utils: NavigationBindingsUtilsObject;
+            public annotationToFields(annotation: Annotation): Dictionary<string>;
+            public bindingsButtonClick(
+                button: HTMLDOMElement,
+                events: (NavigationBindingsOptionsObject|Dictionary<Function>),
+                clickEvent: PointerEventObject
+            ): void;
+            public bindingsChartClick(chart: AnnotationChart, clickEvent: PointerEventObject): void;
+            public bindingsContainerMouseMove(container: HTMLDOMElement, moveEvent: PointerEventObject): void;
+            public deselectAnnotation(): void;
+            public destroy(): void;
+            public fieldsToOptions<T>(fields: PopupFieldsDictionary<string>, config: T): T;
+            public getButtonEvents(container: HTMLDOMElement, event: Event): NavigationBindingsButtonEventsObject;
+            public getClickedClassNames(container: HTMLDOMElement, event: Event): Array<[string, HTMLDOMElement]>;
+            public initEvents(): void;
+            public initUpdate(): void;
+            public removeEvents(): void;
+            public update(options: NavigationOptions): void
+        }
+        interface NavigationBindingsButtonEventsObject {
+            button: HTMLDOMElement;
+            events: NavigationBindingsOptionsObject;
+        }
+        interface NavigationBindingsOptionsObject {
```

Comment:
Not available in API: http://localhost:9005/class-reference/Highcharts.NavigationBindingsOptionsObject

Referenced for example from: http://localhost:9005/highstock/navigation.bindings.arrowInfinityLine

Reply:
I do not understand. Should this internal interface be public or should this not be created internally?

Reply:
I think this should be public. Since here: http://localhost:9005/class-reference/Highcharts.NavigationBindingsOptionsObject - we can click reference to this interface:

![Screen Shot 2020-01-06 at 11 09 58](https://user-images.githubusercontent.com/1453926/71811544-4b3fca00-3075-11ea-9eb2-2a0e1594ff26.png)

I'm pretty sure we users don't want to see this after clicking:
![Screen Shot 2020-01-06 at 11 11 57](https://user-images.githubusercontent.com/1453926/71811608-6e6a7980-3075-11ea-927b-0b3ae9d692bd.png)

Reply:
It should be fixed now.